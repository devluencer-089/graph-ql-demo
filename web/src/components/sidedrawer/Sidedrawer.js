import React from 'react';
import List, { ListItem, ListItemIcon, ListItemText } from 'material-ui/List';
import Divider from 'material-ui/Divider';
import Typography from 'material-ui/Typography';
import UsersIcon from 'material-ui-icons/People';
import { withStyles } from 'material-ui/styles';
import { Link } from 'react-router-dom';

const styles = () => ({
  root: {
    width: '100%',
    maxWidth: 480
  }
});

const listOfMenuEntries = [
  {
    label: 'Alle User',
    icon: <UsersIcon />,
    value: '/users'
  }
];

export const Sidebar = props => {
  const { classes } = props;

  return (
    <div className={classes.root}>
      <List component="nav">
        {listOfMenuEntries.map(entry => (
          <div key={entry.label}>
            <Link to={entry.value} style={{ textDecoration: 'none' }}>
              <ListItem button>
                <ListItemIcon>{entry.icon}</ListItemIcon>
                <ListItemText
                  disableTypography
                  primary={<Typography type="h3">{entry.label}</Typography>}
                />
              </ListItem>
            </Link>
          </div>
        ))}
      </List>
      <Divider />
    </div>
  );
};

export default withStyles(styles)(Sidebar);
