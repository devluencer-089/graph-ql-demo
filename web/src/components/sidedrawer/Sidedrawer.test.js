import React from 'react';
import { shallow } from 'enzyme';
import Sidedrawer from './Sidedrawer';

describe('Sidedrawer', () => {

    it('should match snapshot', () => {

        let eut = shallow(<Sidedrawer/>);

        expect(eut).toMatchSnapshot();
    });
});
