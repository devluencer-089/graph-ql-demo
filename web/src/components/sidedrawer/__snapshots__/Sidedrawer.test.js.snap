// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppBar should match snapshot 1`] = `
<Component
  classes={
    Object {
      "root": "Component-root-1",
    }
  }
/>
`;
